### PR TITLE
Abstraction around raw transaction status

### PR DESCRIPTION
**Pull request by keefertaylor**
`XpringClientDecorator` will share a definition across the legacy protocol buffers (from `xpring-common-protocol-buffers`) and the rippled protocol buffers. 

This PR creates an abstraction that wraps the return type from this method in an interface. This will allow us to create a bridge layer between the old and new protocol buffers during the migration.

# Background

rippled is going to implement protocol buffer support natively, which is going to use a set of well reviewed protocol buffers. The implementation in rippled is being done here: https://github.com/ripple/rippled/pull/3159.

In order to do the upgrade in place, we will swap out the base decorator with two implementations: 
- `LegacyDefaultXpringClient`: Uses the old protocol buffers
- `DefaultXpringClient`: Uses the new protocol buffers

Once we have a working implementation, the constructor on `XpringClient` will be changed to take a new boolean parameter, `useLegacy`, which will control with base implementation is used. 

# Status
- [x] [Generate new versions of the protocol buffers](https://github.com/xpring-eng/Xpring-JS/pull/108)
- [x] [Migrate existing code to a `legacy` subdirectory and namespace](https://github.com/xpring-eng/Xpring-JS/pull/109)
- [ ] Create code that works with the new protocol buffers from the rippled team
   - [x] Create a new `NetworkClient` with fakes (this PR)
   - [ ] Create a new `DefaultXpringClient` with tests
   - [ ] Read only calls for new `DefaultXpringClient`
   - [ ] Write calls fro `DeafultXpringClient`
- [ ] Add a boolean parameter to switch between implementations

### CODE REVIEW

File: src/legacy/legacy-default-xpring-client.ts [hunk 1]
```diff
@@ -7,7 +7,8 @@ import { GetTransactionStatusRequest } from "../../generated/legacy/get_transact
 import { Payment } from "../../generated/legacy/payment_pb";
 import { SubmitSignedTransactionRequest } from "../../generated/legacy/submit_signed_transaction_request_pb";
 import { Transaction } from "../../generated/legacy/transaction_pb";
-import { TransactionStatus as RawTransactionStatus } from "../../generated/legacy/transaction_status_pb";
+import { TransactionStatus as RawTransactionStatusResponse } from "../../generated/legacy/transaction_status_pb";
```
Comment (hbergren) [inline src/legacy/legacy-default-xpring-client.ts]: I don't see anywhere in the PR diff that `RawTransactionStatusResponse` is used. Do we need this import?

**Comment by keefertaylor** [comment]
> My general question is for my own understanding. It looks like before, we were both importing RawTransactionStatus from xpring-common-js and from ../../generated/legacy/transaction_status_pb, and now Xpring-JS is responsible for its own RawTransactionStatus interface.

Yup. So before:
`TransactionStatus`: An enum we return for `getTransactionStatus` calls. 
`RawTransactionStatus`: A raw protocol buffer returned from the server that represents a transaction status

Now:
`TransactionStatus`: same as before
`RawTransactionStatus`: An interface which the previous `RawTransactionStatus` protocol buffer happens to conform to. 

> That interface is our bridge from the old world to the new, so clients can just use the interface and not worry about the underlying implementation. Is that all correct?

Correct. Now the `XpringClientDecorator` interface is decoupled from any protocol buffer implementation so we can move between them.

**Comment by keefertaylor** [comment]
> Where is the transaction's outcome (transaction metadata) accessible?
It is not.

**Comment by intelliot** [comment]
Transaction metadata is necessary even for payments only. If someone sends you a partial payment, you need to look at the metadata in order to know how much money the payment gave you.

**Pull request merged**